src/components/ProcessCreate/Steps/Confirm.tsx

```diff
@@ -101,24 +101,16 @@ export const Confirm = () => {
       }
       const election = Election.from(params)
 
-      let pid: string = ''
-      if (census instanceof CspCensus) {
-        pid = await client.electionService.nextElectionId(account!.address, election)
-        const createdCspElection: IElectionWithTokenResponse = await createElectionInCsp(pid, form.userList)
-      }
-
       for await (const step of client.createElectionSteps(election)) {
         switch (step.key) {
           case ElectionCreationSteps.CENSUS_CREATED:
           case ElectionCreationSteps.SIGN_TX:
           case ElectionCreationSteps.DONE:
             setStep(step.key as Steps)
             if (step.key === ElectionCreationSteps.DONE) {
-              if (pid !== step.electionId) {
-                pid = step.electionId
-                if (census instanceof CspCensus) {
-                  const createdCspElection: IElectionWithTokenResponse = await createElectionInCsp(pid, form.userList)
-                }
+              const pid = step.electionId
```

Comment:
Careful. Doing it this way, the `setCreated` below won't know the actual pid assigned here, since you created a new scope.